apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ExtensionInjector.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.context;
+
+/**
+ * Inject or read the extension protocol fields,such as {@link ExtensionContext#sendingTimestamp}.
+ */
+public class ExtensionInjector {
+
+    private final ExtensionContext extensionContext;
+
+    ExtensionInjector(final ExtensionContext extensionContext) {
+        this.extensionContext = extensionContext;
+    }
+
+    public void injectSendingTimestamp() {
+        extensionContext.setSendingTimestamp(System.currentTimeMillis());
+    }
+
+    public long readSendingTimestamp() {
```

Comment:
`long` -> `Long`, as it could be NULL if don't exist in the in-wire context.

Reply:
This comment is not addressed yet, I mark it as unresolved. @EvanLjp please only mark the conversation as "resolved" after you have actually resolved it or have reasons not to resolve, I don't see either.

Because `sendingTimestamp` is `Long`, returning it from `public long readSendingTimestamp()` will lead to an auto-unboxing and `NullPointerException`